lib/JSON/Validator.pm

```diff
@@ -394,7 +394,7 @@ sub _resolve {
   $self->{level}++;
   $self->_register_schema($schema, $id);
 
-  my @topics = ([$schema, Mojo::URL->new($id)]);
+  my @topics = ([$schema, blessed($id)//'' eq 'Mojo::File' ? $id : Mojo::URL->new($id)]);
```

Comment:
Could you do `UNIVERSAL::isa($id, 'Mojo::File')` instead? Also, this does not look like it's been run through perltidy.